components/automate-cli/pkg/diagnostics/option.go

```diff
@@ -0,0 +1,75 @@
+package diagnostics
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+var ErrInvalidOption = errors.New("Invalid option")
+
+type Option struct {
+	Key   string
+	Value string
+}
+
+func OptionsFromStrings(options []string) ([]*Option, error) {
+	out := make([]*Option, len(options))
+	for i, o := range options {
+		opt, err := OptionFromString(o)
+		if err != nil {
+			return nil, errors.Wrapf(err, "%q is not a valid option", o)
+		}
+		out[i] = opt
+	}
+	return out, nil
+}
+
+func OptionFromString(option string) (*Option, error) {
+	parts := strings.SplitN(option, ":", 2)
+	if len(parts) == 0 {
+		return nil, ErrInvalidOption
+	}
+	return &Option{
+		Key:   parts[0],
+		Value: parts[1],
+	}, nil
+}
+
+func (o *Option) GetString(defaultVal string) string {
+	if o == nil || o.Value == "" {
+		return defaultVal
+	}
+	return o.Value
+}
+
+func (o *Option) GetInt(defaultVal int) int {
+	if o == nil || o.Value == "" {
+		return defaultVal
+	}
+	i, err := strconv.Atoi(o.Value)
+	if err != nil {
+		logrus.WithError(err).Warnf("could not parse int (%s)", o.Value)
```

Comment:
Not terribly in love with a method that can return an error swallowing it instead of returning it.